fft/src/KokkosFFT_Host_transform.hpp

```diff
@@ -9,40 +9,41 @@
 
 namespace KokkosFFT {
 namespace Impl {
-template <typename PlanType, typename... Args>
-void exec_plan(PlanType& plan, float* idata, fftwf_complex* odata,
-               int /*direction*/, Args...) {
-  fftwf_execute_dft_r2c(plan, idata, odata);
+
+template <typename ScopedPlanType>
```

Comment:
Can we also replace this template parameter by`ScopedFFTWPlan` ?

Reply:
It is not that easy.
`ScopedFFTWPlan <ExecSpace, T1, T2>` is a class template, where `ExecSpace ` can be either `Kokkos::Serial`, `Kokkos::Threads`, or `Kokkos::OpenMP`
To make the matter worse, these `T1` and `T2` are not the types of `idata` and `odata`. 
Thus, I think it is better to use template here

Reply:
We could still use templates but for the plan:
```
template <typename ExecutionSpace, typename T1, typename T2>
void exec_plan(ScopedFFTWPlan<ExecutionSpace, T1, T2> const& scoped_plan, float* idata, fftwf_complex* odata,
               int /*direction*/) {
  fftwf_execute_dft_r2c(scoped_plan.plan(), idata, odata);
}
```

This way we would know better what input is expected don't you think ?